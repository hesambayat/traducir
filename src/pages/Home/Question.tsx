import { Shuffle } from '../../componentes'

interface QuestionProps {
  answers: number
  origin: string
  transforming: boolean
}

const Question = ({ answers, origin, transforming = false }: QuestionProps) => {
  return (
    <h1 className="home__question">
      {answers > 1 ? `What are ${answers} Spanish words for` : `What's Spanish for`} <Shuffle text={origin} live={!!transforming} />
    </h1>
  )
}

export default Question
