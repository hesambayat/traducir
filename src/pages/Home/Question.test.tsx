import { render } from '@testing-library/react'
import Question from './Question'

describe('renders home page feature, question', () => {
  const { asFragment } = render(<Question answers={3} origin={'do'} transforming={false} />)
  const dom = asFragment()
  test('should match the snapshot', () => {
    expect(dom).toMatchSnapshot()
  })
})
