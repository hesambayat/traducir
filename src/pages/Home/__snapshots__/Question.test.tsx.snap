// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home page feature, question should match the snapshot 1`] = `
<DocumentFragment>
  <h1
    class="home__question"
  >
    What are 3 Spanish words for 
    <span
      class="shuffle"
    >
      do
    </span>
  </h1>
</DocumentFragment>
`;
